moto/sagemaker/models.py

```diff
@@ -0,0 +1,364 @@
+from __future__ import unicode_literals
+from datetime import datetime
+
+from .exceptions import MissingModel
+from moto.ec2 import ec2_backends
+from moto.ecr.models import BaseObject
+from moto.core import BaseBackend
+from moto.core.exceptions import RESTError
+from moto.sagemaker import validators
+from moto.sts.models import ACCOUNT_ID
+
+
+class Model(BaseObject):
+    def __init__(
+        self,
+        region_name,
+        model_name,
+        execution_role_arn,
+        primary_container,
+        vpc_config,
+        containers=[],
+        tags=[],
+    ):
+        self.ModelName = model_name
+        self.creation_time = datetime.now().strftime("%Y-%m-%d %H:%M:%S")
+        self.Containers = containers
+        self.Tags = tags
+        self.EnableNetworkIsolation = False
+        self.VpcConfig = vpc_config
+        self.PrimaryContainer = primary_container
+        self.ExecutionRoleArn = execution_role_arn or "arn:test"
+        self.ModelArn = self.arn_for_model_name(self.ModelName, region_name)
```

Comment:
Variables should be in `snake_case_format`, i.e. no upper cases and underscores in between words

(Edit: Goes for all models, VpcConfig and Container as well)

Reply:
The rationale behind this non-pythonic style is to match the case of the API response, such that a simple generic mechanism for generating the response can be utilized.  I believe @porthunt adapted this technique from the ECR models.

While I'm certainly a believer in consistent coding style, I favor simpler, shorter code and fewer transformations over that.

If you still feel strongly about this, would you point me toward the moto module you feel implements simple response generation from snake_case model attributes?  Tx.

Reply:
ECR keeps snake_case, actually and only generates camelCase on request: https://github.com/spulec/moto/blob/master/moto/ecr/models.py#L27

Reply:
Feel free to just copy paste the relevant bits btw - it's used in ECS as well, so I'd be happy to refactor these methods into a utility method/class afterwards

Reply:
Thanks for the pointer and clarification Bert, will do.

Reply:
Bert, for your notes wrt ECR's BaseObject's camel-caser for future refactoring:
- It intentionally leaves the key's first char lowercase, as appropriate for ECR, but not for Sagemaker.  I had to modify it.
- It leaves keys without underscores unmodified.  In Sagemaker I need them to be capitalized.

Reply:
Done.